api/api-pipeline-groups-v1/src/main/java/com/thoughtworks/go/apiv1/admin/pipelinegroups/PipelineGroupsControllerV1.java

```diff
@@ -148,10 +148,15 @@ public String update(Request req, Response res) {
     }
 
     public String destroy(Request req, Response res) throws IOException {
-        PipelineConfigs pipelineConfigs = fetchEntityFromConfig(req.params("group_name"));
+        String groupName = req.params("group_name");
+        PipelineConfigs pipelineConfigs = fetchEntityFromConfig(groupName);
 
         HttpLocalizedOperationResult result = new HttpLocalizedOperationResult();
-        pipelineConfigsService.deleteGroup(SessionUtils.currentUsername(), pipelineConfigs, result);
+        if (!pipelineConfigs.isEmpty()) {
+            result.unprocessableEntity(String.format("Failed to delete group %s because it was not empty.", groupName));
+        } else {
+            pipelineConfigsService.deleteGroup(SessionUtils.currentUsername(), pipelineConfigs, result);
+        }
```

Comment:
@kritika-singh3 @viraj2712 -- How about performing this config entity related validation under [`DeletePipelineConfigsCommand.isValid`](https://github.com/gocd/gocd/blob/master/server/src/main/java/com/thoughtworks/go/config/update/DeletePipelineConfigsCommand.java#L45-L47)?

Reasons:
1. Controllers are not responsible for validations, they should just call the service methods.
2. This validation will not be performed for anyone directly calling `pipelineConfigsService.deleteGroup` method.

Reply:
Agree that the current implementation leaves some margin of error. Moving the empty check to `DeletePipelineConfigsCommand.canContinue`.